### ISSUE TITLE
ASP.NET Deadlocks when using .Result because of missing ConfigureAwait(false)

### ISSUE THREAD

**Issue by rikvandenberg** [voucherifyio/voucherify-dotNET-sdk#12]
Hi All,

I'm creating the issue to outline some of our problems we are having in ASP.NET WebAPI and using the .NET library you have provided. We have the following code:

```
public VoucherModel GetVoucherAsync(string voucherCode)
{
    Voucher voucher = _api.Vouchers.Get(voucherCode).Result;
    return Mapper.Map<VoucherModel>(voucher);
}
```
We use `.Result` here because our code doesn't support asynchronous operations just yet. And this [blog](https://blogs.msdn.microsoft.com/alazarev/2017/05/20/fun-with-configureawait-and-deadlocks-in-asp-net/) outlines and explaines it what is causing the deadlock. To quote:

> So, according to it, ASP.NET has a specialized sync context, that only one thread can have. And if you don’t use .ConfigureAwait(false), it tries to restore old context, which belongs to the main thread that is blocked by .Result, hence deadlock...

This issue will not show up in the example because none of those examples are in ASP.NET 

I will create an PR to solve this issue. :smile:

**Comment by bandraszyk** [comment]
Thank you for reporting and fixing the issue. I will verify your changes later today or tomorrow morning. If all will be good then we will merge and release new version.

**Comment by bandraszyk** [comment]
Your branch has been merged and we have released new version: `4.1.1`. Should be indexed soon and available to be used.
